### ISSUE TITLE
Support Unit L1 Ball constraint

### ISSUE THREAD

**Issue by fehiepsi** [pyro-ppl/pyro#2862]
This constraint is useful for the `skewness` parameter of Sine Skewed distribution introduced in #2826.

### Motivation

It is nontrivial to parameterize the skewness parameter. For example, even in case d=2, currently we need 4 non-trivial lines of code
```python
    skew_phi = pyro.sample('skew_phi', Uniform(-1., 1.))
    psi_bound = 1 - skew_phi.abs()
    skew_psi = pyro.sample('skew_psi', Uniform(-1., 1.)
    skewness = torch.stack((skew_phi, psi_bound * skew_psi), dim=-1)
```

### Approach

We can use [signed stick breaking transform](https://github.com/pytorch/pytorch/blob/94cc681fc2e5218c10493938a8ca01272c3c6fc0/torch/distributions/transforms.py#L763):
    + Scales into the interval `(-1, 1)` domain: `r = tanh(X)` (modulo scaling, we can use `2*sigmoid(X) - 1`).
    + Transforms into an unsigned domain: `z = abs(r)`.
    + Applies `s = StickBreakingTransform(z)` (this transforms a vector in R^n to R^n+1, but we can ignore the last item because we just need a ball in R^n, not a sphere in R^n+1)
    + Transforms back into signed domain: `y = sign(r) * s`.

cc @OlaRonning
